geopandas/tools/clip.py

```diff
@@ -110,10 +135,15 @@ def clip(gdf, mask, keep_geom_type=False):
             "'gdf' should be GeoDataFrame or GeoSeries, got {}".format(type(gdf))
         )
 
-    if not isinstance(mask, (GeoDataFrame, GeoSeries, Polygon, MultiPolygon)):
+    if not isinstance(mask, (GeoDataFrame, GeoSeries, Polygon, MultiPolygon, tuple)):
```

Comment:
Can we be a bit more flexible here? I can imagine that a lot of people will try things like
```py
gdf.clip(gdf2.total_bounds)
```
But that doesn't work right now. I would try to check for array-like of a length 4 and cast it to a tuple.

Reply:
You can use `pandas.api.types.is_list_like` for that check.

Reply:
Done